modules/tensor_mechanics/doc/content/documentation/systems/Modules/TensorMechanics/GeneralizedPlaneStrain/index.md

```diff
@@ -1,42 +1,54 @@
-# GeneralizedPlaneStrain Action
+# Generalized Plane Strain Action System
 
-Sets up a generalized plane strain model. This action can be generated using the [TensorMechanics Master action](../Master). A detailed description of generalized plane strain model can be found in the [formulation](tensor_mechanics/generalized_plane_strain.md) page.
+!syntax description /Modules/TensorMechanics/GeneralizedPlaneStrain/GeneralizedPlaneStrainAction
 
-## Example Input File Syntax
+## Description
 
-!listing modules/tensor_mechanics/test/tests/generalized_plane_strain/generalized_plane_strain_squares.i block=Modules/TensorMechanics/GeneralizedPlaneStrain
+This action sets up a generalized plane strain model. A detailed description of generalized plane strain model can be found in the [formulation](tensor_mechanics/generalized_plane_strain.md) page.
+
+!alert warning title=For 2D and 1D Simulations
+GeneralizedPlaneStrain only works for 1D axisymmetric or 2D generalized plane strain cases where the number of displacement variables is only one or two displacements.
+
+## Constructed MooseObjects
+
+The `GeneralizedPlaneStrain` Action is used to construct the kernels, user objects, displacement and scalar out-of-plane variables, and strain materials in a consistent manner as required for a generalized plane strain simulation.
+
+!table id=generalizedPS_action_table caption=Correspondence Among Action Functionality and MooseObjects for the `GeneralizedPlaneStrain` Action
+| Functionality     | Replaced Classes   | Associated Parameters   |
+|-------------------|--------------------|-------------------------|
+| In-plane equilibrium conditions for the given coordinate system | [StressDivergenceTensors](/Kernels/StressDivergenceTensors.md) or [StressDivergenceRZTensors](/Kernels/StressDivergenceRZTensors.md) or [StressDivergenceRSphericalTensors](/Kernels/StressDivergenceRSphericalTensors.md) | `displacements` : a string of the displacement field variables |
```

Comment:
Do the generalized plane strain assumptions apply to an spherically symmetric problem? I think we only have that for the XY and RZ problems.